app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Posts} from 'mattermost-redux/constants';
+import Markdown from 'app/components/markdown';
+
+const renderUsername = (value) => {
+    return (value[0] === '@') ? value : `@${value}`;
+};
+
+const formattedMessage = (localeHolder, elements, intl) => {
+    const {formatMessage} = intl;
+
+    return formatMessage(localeHolder, elements);
+};
+
+const renderHeaderChangeMessage = (postBodyProps, styles, intl) => {
+    const {onPress, postProps} = postBodyProps;
+    const {messageStyle, textStyles} = styles;
+
+    if (!postProps.username) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const oldHeader = postProps.old_header;
+    const newHeader = postProps.new_header;
+    let localeHolder;
+
+    if (postProps.new_header) {
+        if (postProps.old_header) {
+            localeHolder = {
+                id: 'api.channel.post_update_channel_header_message_and_forget.updated_from',
+                defaultMessage: '{username} updated the channel header from: {oldHeader} to: {newHeader}',
+            };
+
+            return (
+                <Markdown
+                    baseTextStyle={messageStyle}
+                    onPostPress={onPress}
```

Comment:
I think you are missing `onLongPress`

Reply:
Hmm... the existing, current implementation of these system markdown messages doesn't have any long press behaviour for messages where the current user isn't the actor. This behaviour is maintained in the new implementation.


| Current Long Press Behaviour | New Long Press Behaviour |
| --- | --- |
| <img width="373" alt="current" src="https://user-images.githubusercontent.com/887849/66723075-558e5700-edeb-11e9-9ee1-6d9900d38de8.gif"> | <img width="372" alt="new" src="https://user-images.githubusercontent.com/887849/66723087-663ecd00-edeb-11e9-9d1a-0361748ed31d.gif">







Please advise if this not desired?

Reply:
Actually, I think the menu is available based on who made the change. You get the long press menu on the second post because you're logged in as `userii`.

Regardless, it seems to be working as intended even without the extra `onLongPress` prop.